components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/GeckoEngine.kt

```diff
@@ -159,35 +160,67 @@ class GeckoEngine(
         onSuccess: ((WebExtension) -> Unit) = { },
         onError: ((String, Throwable) -> Unit) = { _, _ -> }
     ) {
-        if (ext.supportActions) {
-            // We currently have to install the global action handler before we
-            // install the extension which is why this is done here directly.
-            // This code can be removed from the engine once the new GV addon
-            // management API lands. Then the global handlers will be invoked
-            // with the latest state whenever they are registered:
-            // https://bugzilla.mozilla.org/show_bug.cgi?id=1599897
-            // https://bugzilla.mozilla.org/show_bug.cgi?id=1582185
-            ext.registerActionHandler(object : ActionHandler {
-                override fun onBrowserAction(extension: WebExtension, session: EngineSession?, action: BrowserAction) {
-                    webExtensionDelegate?.onBrowserActionDefined(extension, action)
-                }
+        val actionHandler = object : ActionHandler {
+            override fun onBrowserAction(extension: WebExtension, session: EngineSession?, action: BrowserAction) {
+                webExtensionDelegate?.onBrowserActionDefined(extension, action)
+            }
 
-                override fun onToggleBrowserActionPopup(
-                    extension: WebExtension,
-                    action: BrowserAction
-                ): EngineSession? {
-                    val session = GeckoEngineSession(runtime)
-                    return webExtensionDelegate?.onToggleBrowserActionPopup(extension, session, action)
-                }
+            override fun onToggleBrowserActionPopup(extension: WebExtension, action: BrowserAction): EngineSession? {
+                return webExtensionDelegate?.onToggleBrowserActionPopup(extension, GeckoEngineSession(runtime), action)
+            }
+        }
+
+        if (Uri.parse(ext.url).scheme == "resource") {
```

Comment:
Btw,  adding `isBuiltIn` to the properties of the extension is going to be part of the follow up?

Reply:
@Amejia481 hm let me add it here, I think that's easy and we'll need it.

Reply:
Thanks, I will use it on the update part :)